### PR TITLE
feat: add ns discount extention

### PR DESCRIPTION
**Pull request by donaldxdonald**
## Description

Added a new extension for listing discounts infomation of games on Nintendo Switch

## Screencast

image
image
image

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/ns-discount/src/PriceList.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    if (text.length === 0) {
+      setShowPrices(prices)
+    } else {
+      setShowPrices(prices.filter((price) => price.country.includes(text)))
+    }
+  }
+
+  return (
+    <List onSearchTextChange={onSearchTextChanged}>
+      <List.Item key="date" title="截止日期" accessories={[{ text: game.leftDiscount.replace('天', ' 天') }]} />
```
Comment (pernielsentikaer) [inline extensions/ns-discount/src/PriceList.tsx:33]: At the moment, Raycast doesn't support localization and only supports US English, please keep it in English 😄

File: extensions/ns-discount/src/index.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+          <List.Item
+            id={game.appid}
+            key={game.appid}
+            title={game.titleZh}
+            subtitle={game.title}
+            icon={{ source: game.icon }}
+            accessories={[{ text: `${calcCutoff(game.cutoff)}折` }]}
+            actions={
+              <ActionPanel>
+                <ActionPanel.Section title="功能">
```
Comment (pernielsentikaer) [inline extensions/ns-discount/src/index.tsx:38]: At the moment, Raycast doesn't support localization and only supports US English, please keep it in English 😄

File: extensions/ns-discount/src/index.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    <List>
+      {gameList.map((game) => {
+        return (
+          <List.Item
+            id={game.appid}
+            key={game.appid}
+            title={game.titleZh}
+            subtitle={game.title}
+            icon={{ source: game.icon }}
+            accessories={[{ text: `${calcCutoff(game.cutoff)}折` }]}
```
Comment (pernielsentikaer) [inline extensions/ns-discount/src/index.tsx:35]: At the moment, Raycast doesn't support localization and only supports US English, please keep it in English 😄

**Comment by donaldxdonald** [comment]
> Hi @donaldxdonald 👋
> 
> Thanks for your contribution 💪
> 
> I left some feedback for you to address 😄
> 
> * I modified .prettierrc so be as the default for Raycast extensions, you might need to run prettier again for all files and push again.
> * I added proper isLoading property to your list since it was having a weird [empty state flicker](https://developers.raycast.com/basics/prepare-an-extension-for-store#empty-states)
> * I added gracefully error handling instead of just dying hard.
> 
> I'm looking forward to testing this extension again 🔥
> 
> Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

Thanks for your advice. I'll update soon.

**Comment by pernielsentikaer** [comment]
Hi @donaldxdonald 👋 

Are you still around and want to finish this?

**Comment by donaldxdonald** [comment]
> Hi @donaldxdonald 👋
> 
> Are you still around and want to finish this?

Yes. But sorry, I am still looking for another English Ver. API alternative 😂

**Comment by pernielsentikaer** [comment]
I'm closing this until you're ready; feel free to open it again or comment in the thread when you're ready to continue working on it 🙂

You can also catch me here or in [Slack](https://www.raycast.com/community) if you want to discuss 😄

**Pull request closed without merging**